### PR TITLE
Fix more low-hanging Coverity warnings

### PR DESCRIPTION
**Pull request by jmillikin-stripe**
This PR fixes ~10 warnings out of 22. The rest are a little tricky, or seem like Coverity bugs.

### CODE REVIEW

File: source/common/router/rds_subscription.cc [hunk 1]
```diff
@@ -59,6 +59,7 @@ void RdsSubscription::parseResponse(const Http::Message& response) {
 void RdsSubscription::onFetchComplete() {}
 
 void RdsSubscription::onFetchFailure(const EnvoyException* e) {
+  RELEASE_ASSERT(callbacks_ != nullptr);
```
Comment (mattklein123) [inline source/common/router/rds_subscription.cc]: Generally we don't do this as it will crash in an obvious way on the next line. Why is this required?

File: source/common/ssl/context_impl.cc [hunk 2]
```diff
@@ -391,8 +391,10 @@ ServerContextImpl::ServerContextImpl(ContextManagerImpl& parent, Stats::Scope& s
            int encrypt) -> int {
           ContextImpl* context_impl = static_cast<ContextImpl*>(
               SSL_CTX_get_ex_data(SSL_get_SSL_CTX(ssl), sslContextIndex()));
-          return dynamic_cast<ServerContextImpl*>(context_impl)
-              ->sessionTicketProcess(ssl, key_name, iv, ctx, hmac_ctx, encrypt);
+          ServerContextImpl* server_context_impl = dynamic_cast<ServerContextImpl*>(context_impl);
+          RELEASE_ASSERT(server_context_impl != nullptr);
```
Comment (mattklein123) [inline source/common/ssl/context_impl.cc]: this will crash on the following line. why?

File: source/common/router/rds_subscription.cc (hunk 1 above)
Comment (jmillikin-stripe) [reply source/common/router/rds_subscription.cc]: I like to put in asserts like this to force the crash to happen as far up the call graph as possible.

Crashes from null dereference don't always localize well -- the final SIGSEGV can be triggered somewhere deep down in the stack, if no instruction needed to resolve `*this` immediately.

File: source/common/ssl/context_impl.cc (hunk 2 above)
Comment (jmillikin-stripe) [reply source/common/ssl/context_impl.cc]: This one is specifically to fix Coverity's requirement that the return value of `dynamic_cast` be checked for null-ness.

File: source/common/router/rds_subscription.cc (hunk 1 above)
Comment (mattklein123) [reply source/common/router/rds_subscription.cc]: In general this has not been my experience, and we don't currently do this anywhere else (see error handling / style guide). For consistency I would prefer to not do this. If we need it for coverity can add relevant comments?

Comment (jmillikin-stripe) [reply source/common/router/rds_subscription.cc]: OK, removed these asserts of non-null callbacks, and added a comment to the dynamic_cast assertion.

**Comment by mattklein123** [comment]
@jmillikin-stripe LGTM. Can you fix DCO?

**Comment by jmillikin-stripe** [comment]
DCO fixed.

Is there some way to get that `signed-off-by` magic text added automatically, or skip the check for committers with a CLA on file? I don't understand why it's different from the CLA we spent ~weeks getting the lawyers to sign.

**Comment by mattklein123** [comment]
cc @caniszczyk re: DCO workflow pain. We are working with GH to hopefully eventually make the flow better. The reason to use DCO is so that other companies don't have to spend weeks getting the CLA signed. 😉

**Comment by jmillikin-stripe** [comment]
All checks now pass.
